zip-0315.rst

```diff
@@ -1,9 +1,673 @@
+
+  * This may happen because a previous transaction expired and the user is
+    trying to spend some of the same TXOs.
+
+* Whether the transaction is coinbase
+
+* For coinbase transactions, the amounts / destination addresses / memos of
+  shielded outputs
+
+* Orchard flags (enableSpends/enableOutputs)
+
+  * Under normal circumstances these only depend on whether the transaction
+    is coinbase.
+
+
+Linkability of transactions or addresses
+''''''''''''''''''''''''''''''''''''''''
+
+Motivation for choices reducing linkability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We want to support creating unlinkable addresses, in order that a user can
+give different addresses to different counterparties, in such a way that the
+counterparties (even if they collude) cannot tell that the addresses were
+provided by the same or distinct users.
+
+If multiple UTXOs are received at the same transparent address, it is safe
+to shield them all in the same transaction, because that is not leaking
+additional information.
+
+UTXOs received on different transparent receivers SHOULD NOT be shielded
+in the same transaction. Ideally, when they are shielded in separate
+transactions, this should be done in such a way that the timing of those
+transactions is not linkable.
+
+When spending transparent UTXOs, they SHOULD only be sent to an internal
+shielded receiver belonging to the wallet.
+
+A wallet MUST NOT send funds to a transparent address unless all of the
+source funds come from shielded pools, and this SHOULD be a single shielded
+pool.
+
+We want to minimize pool crossing.
+
+Anchor selection
+''''''''''''''''
+
+A wallet SHOULD choose an anchor a number of blocks back from the head of the
+chain equal to the trusted confirmation depth. That is, if the current block
+is at height H, the anchor SHOULD reflect the final treestate of the block at
+height H-3.
+
+
+Rationale for anchor selection
+''''''''''''''''''''''''''''''
+
+* If the chain rolls back past the block at which the anchor is chosen, then
+  the anchor and the transaction will be invalidated. This is undesirable
+  both for reliability, and because the nullifiers of spent shielded notes
+  will have been revealed, linking this transaction to any future transactions
+  that spend those notes.
+* On the other hand, it is undesirable to choose an anchor too many blocks
+  back, because that prevents more recently received shielded notes from
+  being spent.
+* Using a fixed anchor depth (as opposed to a different depth depending on
+  whether or not we are spending trusted shielded notes) avoids leaking
+  information about whether or not the shielded notes we spent were
+  trusted.
+
+
+Note selection
+''''''''''''''
+
+TODO: consider what we should do when nullifiers are revealed in a transaction
+that is then invalidated. Should those notes be prioritized to be spent soon,
+or should they be used in a note management tx?
```

Comment:
```suggestion
TODO: recommend that when nullifiers are revealed in a transaction that is then
invalidated (or revealed in some other detectable way), they SHOULD be used in
a note management tx to avoid linking the invalidated tx with some arbitrary
future tx. Provided that there are no transparent outputs leaving this wallet's
control, the same arities and transparent outputs SHOULD be preserved, which
also avoids revealing whether the user changed their mind about whether to send
the original semantic transaction.
```